bsmetadata/preprocessing_utils.py

```diff
@@ -80,3 +83,86 @@ def _extract_timestamp_from_url(self, url: str) -> Optional[str]:
         date = remove_improbable_date(date)
         date = str(date) if date is not None else None
         return date
+
+
+class GenerationLengthPreprocessor(MetadataPreprocessor):
+    """An exemplary metadata preprocessor for adding generation length information based on text."""
+
+    def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
+        for example_text, example_metadata in zip(examples["text"], examples["metadata"]):
+            example_length = self._extract_length_from_text(example_text)
+
+            if not example_length:
+                continue
+
+            example_metadata.append({"key": "length", "type": "global", "value": example_length})
+
+        return examples
+
+    def _extract_length_from_text(self, text: str) -> Optional[str]:
+        return str(len(text))  # global
+
+
+class DatasourcePreprocessor(MetadataPreprocessor):
+    """An exemplary metadata preprocessor for adding datasource information based on URLs."""
+
+    def _check_numbers(self, sub_part):
```

Comment:
I think readability could be improved here by adding type hints (`_check_numbers(self, sub_part: List[str]) -> List[str]`) but I guess that's a matter of taste. At first, I thought that `sub_part` was a `str` and was confused why you would only remove the first/last character.

Reply:
✅ good point, I like the idea. I've updated the function in this regard. thanks a lot